cypress/e2e/awx/administration/credentialTypes.cy.ts

```diff
@@ -0,0 +1,290 @@
+      cy.createAndDeleteCustomAWXCredentialTypeUI(
+        customCredentialTypeName,
+        inputCredType,
+        injectorCredType,
+        'json'
+      );
+      cy.verifyPageTitle('Credential Types');
+    });
+
+    it('creates a custom credential type with input and injector configurations in YAML format in the Monaco editor', () => {
+      cy.createAndDeleteCustomAWXCredentialTypeUI(
+        customCredentialTypeName,
+        inputCredType,
+        injectorCredType
+      );
+      cy.verifyPageTitle('Credential Types');
+    });
+  });
+
+  describe('Credential Types: Edit and Delete Actions', () => {
+    beforeEach(() => {
+      cy.createAwxCredentialType().then((credentialType) => {
+        credType1 = credentialType;
+      });
+      cy.createAwxCredentialType().then((credentialType) => {
+        credType2 = credentialType;
+      });
+    });
+
+    afterEach(() => {
+      cy.deleteAwxCredentialType(credType1, { failOnStatusCode: false });
+      cy.deleteAwxCredentialType(credType2, { failOnStatusCode: false });
+    });
+
+    it('checks that deleting a custom credential type which is being used by a credential is not allowed', () => {
+      cy.navigateTo('awx', 'credential-types');
+      cy.filterTableBySingleSelect('name', 'Google Compute Engine');
+      cy.selectTableRowByCheckbox('name', 'Google Compute Engine', { disableFilter: true });
+      cy.clickToolbarKebabAction('delete-selected-credential-types');
+      cy.getModal().within(() => {
+        cy.getByDataCy('alert-toaster')
+          .should('be.visible')
+          .and(
+            'contain',
+            '1 of the selected credential types cannot be deleted because it is read-only.'
+          );
+        cy.getBy(
+          '[data-ouia-component-id="Permanently delete credential types-ModalBoxCloseButton"]'
+        ).click();
+      });
+    });
+
+    it('edit a credential type from the list row action and delete it using the list kebab menu', () => {
+      cy.navigateTo('awx', 'credential-types');
+      const editedCredentialTypeName = (credType1.name ?? '') + ' edited';
+      cy.filterTableByMultiSelect('name', [credType1.name]);
+      cy.clickTableRowPinnedAction(credType1.name, 'edit-credential-type', false);
+      cy.verifyPageTitle('Edit Credential Type');
+      cy.url().then((currentUrl) => {
+        expect(currentUrl.includes('edit')).to.be.true;
+      });
+      cy.get('[data-cy="name"]').clear().type(editedCredentialTypeName);
+      cy.get('[data-cy="description"]').clear().type('this is a new description after editing');
+      cy.intercept('PATCH', `api/v2/credential_types/${credType1.id}/`).as('editCredType');
+      cy.clickButton(/^Save credential type$/);
+      cy.wait('@editCredType')
+        .its('response.body.name')
+        .then((name: string) => {
+          expect(editedCredentialTypeName).to.be.equal(name);
+        });
+      cy.verifyPageTitle(editedCredentialTypeName);
```

Comment:
I could not find the assertion related to the edited items. Do we verify that after was saved the details page show the new name and new description?